### DISCUSSION TITLE
How to post a tweet using tweepy?

### DISCUSSION THREAD

**Discussion by yesdeepakmittal** [tweepy/tweepy#1787] [Questions] [answered]
How can I post a tweet using API.update_status()? It needs POST permission but I did not find this app permission option on the Twitter Developer App page.

**Comment by realrodri** [comment]
You need to apply for the "Elevated" Twitter Developer program. It's free, you just need to submit a form and maybe a couple of emails due to Twitter's auto response, but then you should be good to go. I'd recommend then follow this steps:

First, install tweepy library (pip install tweepy).

Next, create a .py file containing this:

```
import tweepy
auth = tweepy.OAuthHandler("CONSUMER KEY HERE", "CONSUMER KEY SECRET HERE")
auth.set_access_token("ACCESS TOKEN HERE", "ACCESS TOKEN SECRET HERE")
api = tweepy.API(auth)
tweet = input(" ")
api.update_status(status =(tweet))
print ("Done!")
```

Your tweet should be live then.

**Comment by bybilal** [comment]
> 

how am i able to post a tweet with only the essential account?

**Reply by Harmon758** [reply]
That's unrelated to this, but you can use [`Client.create_tweet`](https://tweepy.readthedocs.io/en/v4.5.0/client.html#tweepy.Client.create_tweet).

**Comment by Harmon758** [comment, marked as answer]
The OAuth 1.0a App permissions setting is under the User authentication settings section of the Settings tab of your developer portal's App page.
